src/bin/ch-remote.rs

```diff
@@ -652,334 +895,238 @@ fn create_data(path: &str) -> Result<String, Error> {
+                    Arg::new("id")
+                        .long("id")
+                        .help("Memory zone identifier")
+                        .num_args(1),
+                )
+                .arg(
+                    Arg::new("size")
+                        .long("size")
+                        .help("New memory zone size in bytes (supports K/M/G suffix)")
+                        .num_args(1),
+                ),
+        )
+        .subcommand(Command::new("resume").about("Resume the VM"))
+        .subcommand(Command::new("boot").about("Boot a created VM"))
+        .subcommand(Command::new("delete").about("Delete a VM"))
+        .subcommand(Command::new("shutdown").about("Shutdown the VM"))
+        .subcommand(
+            Command::new("snapshot")
+                .about("Create a snapshot from VM")
+                .arg(
+                    Arg::new("snapshot_config")
+                        .index(1)
+                        .help("<destination_url>"),
+                ),
+        )
+        .subcommand(
+            Command::new("restore")
+                .about("Restore VM from a snapshot")
+                .arg(
+                    Arg::new("restore_config")
+                        .index(1)
+                        .help(vmm::config::RestoreConfig::SYNTAX),
+                ),
+        )
+        .subcommand(
+            Command::new("coredump")
+                .about("Create a coredump from VM")
+                .arg(Arg::new("coredump_config").index(1).help("<file_path>")),
+        )
+        .subcommand(
+            Command::new("send-migration")
+                .about("Initiate a VM migration")
+                .arg(
+                    Arg::new("send_migration_config")
+                        .index(1)
+                        .help("<destination_url>"),
+                )
+                .arg(
+                    Arg::new("send_migration_local")
+                        .long("local")
+                        .num_args(0)
+                        .action(ArgAction::SetTrue),
+                ),
+        )
+        .subcommand(
+            Command::new("receive-migration")
+                .about("Receive a VM migration")
+                .arg(
+                    Arg::new("receive_migration_config")
+                        .index(1)
+                        .help("<receiver_url>"),
+                ),
+        )
+        .subcommand(
+            Command::new("create")
+                .about("Create VM from a JSON configuration")
+                .arg(Arg::new("path").index(1).default_value("-")),
+        )
+        .subcommand(Command::new("ping").about("Ping the VMM to check for API server availability"))
+        .subcommand(Command::new("shutdown-vmm").about("Shutdown the VMM"));
 
-    if matches!(toplevel.command, SubCommandEnum::Version(_)) {
-        println!("{} {}", env!("CARGO_BIN_NAME"), env!("BUILD_VERSION"));
-        return;
-    }
+    let matches = app.get_matches();
```

Comment:
The `version` command is dropped and needs to be added back. 

One tip: you will need to get rid of the `.subcommand_required(true)` and handle the `version` case manually and return early.

Reply:
I think `.version(env!("BUILD_VERSION"))` is enough.

Reply:
It's better to provide the same version information from `ch-remote` as Cloud Hypervisor, so that users can cross check they are using the same versions for both binaries (to ensure the APIs are from the same version). I believe this is why the `version` command is handled manually.

Reply:
version handled in ch-remote with `.version(env!("BUILD_VERSION"))`
```
$ target/debug/ch-remote --version
ch-remote v35.0-35-gc940027a-dirty
```

version handled in cloud-hypervisor binary manually
```
    if cmd_arguments.get_flag("version") {
        println!("{} {}", env!("CARGO_BIN_NAME"), env!("BUILD_VERSION"));
    }
```
```
$ target/debug/cloud-hypervisor --version
cloud-hypervisor v35.0-35-gc940027a-dirty
```

in both the places we are accessing though `env!("BUILD_VERSION")`, so it is printing same information. am I missing something here?

Reply:
You are right. I missed the fact that you are already using `BUILD_VERSION`. Looks like the manual handling on the `version` command was from using `argh` and we need to keep it to support `--version -v`. 

PR now looks good to me. I will rerun the testing PR and assume everything is good I will approve it.